support/util/imagefetch.go

```diff
@@ -0,0 +1,63 @@
+package util
+
+import (
+	"context"
+	"fmt"
+	"github.com/blang/semver/v4"
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+	"github.com/openshift/hypershift/support/releaseinfo"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+// GetHypershiftComponentImage resolves the appropriate control plane operator
+// image based on the following order of precedence (from most to least
+// preferred):
+//
+// 1. The image specified by the ControlPlaneOperatorImageAnnotation on the
+//    HostedControlPlane resource itself
+// 2. The hypershift image specified in the release payload indicated by the
+//    HostedControlPlane's release field
+// 3. The hypershift-operator's own image for release versions 4.9 and 4.10
+//
+// If no image can be found according to these rules, an error is returned.
+func GetHypershiftComponentImage(ctx context.Context, objectAnnotations map[string]string, releaseImage string, releaseProvider releaseinfo.Provider, hypershiftOperatorImage string, pullSecret []byte) (string, error) {
+	if val, ok := objectAnnotations[hyperv1.ControlPlaneOperatorImageAnnotation]; ok {
+		return val, nil
+	}
+	releaseInfo, err := releaseProvider.Lookup(ctx, releaseImage, pullSecret)
+	if err != nil {
+		return "", err
+	}
+	version, err := semver.Parse(releaseInfo.Version())
+	if err != nil {
+		return "", err
+	}
+
+	if hypershiftImage, exists := releaseInfo.ComponentImages()["hypershift"]; exists {
+		return hypershiftImage, nil
+	}
+
+	versionMajMin := fmt.Sprintf("%d.%d", version.Major, version.Minor)
+	switch versionMajMin {
+	case "4.9", "4.10":
+		return hypershiftOperatorImage, nil
```

Comment:
4.10 should be always coming from the payload, i.e `cpoImage(HC.spec.release)`

Reply:
I think people will still want this as a test path

Reply:
so I do feel like the precedence is beneficial

Reply:
so for now my preference is to leave this as is for same behavior and a future pr can specifically remove the 4.10 piece if that is desired